avp.c

```diff
@@ -515,9 +505,24 @@ int result_code_avp (struct tunnel *t, struct call *c, void *data,
         return 0;
     }
 
+    if (datalen >= 8) {
+        error = ntohs (raw[4].s);
+        if (((error & 0xFF) == 0) && (error >> 8 != 0))
+        {
+            if (DEBUG)
+                l2tp_log (LOG_DEBUG,
+                     "%s: error code endianness fix for buggy Apple client. network=%d, le=%d\n",
+                     __FUNCTION__, error, error >> 8);
+            error >>= 8;
+        }
+    }
+
     c->error = error;
     c->result = result;
-    safe_copy (c->errormsg, (char *) &raw[5].s, datalen - 10);
+    if (datalen > 10)
```

Comment:
Just want to make sure this is intended to be for _datalen > 10_ (and not _datalen > 8_)

Reply:
Hm good catch, I think the issue is actually in line 508, it should be: if (datalen >= 10)

See for reference: https://tools.ietf.org/html/rfc2661#section-4.4.2

I'll submit a new version of the patch